library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -51,4 +62,35 @@ object JSConverters {
       arr: scala.Array[T]): JSRichGenTraversableOnce[T] =
     new JSRichGenTraversableOnce(arr)
 
+  @inline
+  implicit def JSRichFutureThenable[A](f: Future[Thenable[A]]): JSRichFuture[A] =
+    new JSRichFuture[A](f.asInstanceOf[Future[A | Thenable[A]]])
```

Comment:
`Thenable[Thenable[A]]` is unsound because, assuming you have a `t` of that type, then in `t.then(x => e)`, `x` will in fact be of type `A` (because of flattening). But the type signature of `Thenable.then` says that `x` should be `Thenable[A]`.

Another way to look at it is: by construction, there cannot be any `t: Thenable[Thenable[A]]`. Any attempt to construct a `Thenable[Thenable[A]]` would in fact result in a `Thenable[A]` instead, because of flattening.